index.d.ts

```diff
@@ -0,0 +1,267 @@
+import EventEmitter from 'events';
+import { PassThrough, Writable } from 'stream';
+
+type EventsType = {
+  PASSED: 'passed';
+  FAILED: 'failed';
+  SKIPPED: 'skipped';
+  STARTED: 'started';
+  FINISHED: 'finished';
+}
+
+type zUnitOptions = Partial<{
+  defaults: unknown;
+  initial: unknown;
+  runtime: unknown;
+  bequeathed: unknown;
+}>;
+
+declare class Runnable extends EventEmitter.EventEmitter {
+  constructor(name: string);
+
+  name: string;
+
+  description: string;
+}
+
+type zUnitSyntax = {
+  describe: typeof describe;
+  xdescribe: typeof xdescribe;
+  odescribe: typeof odescribe;
+  it: typeof it;
+  xit: typeof xit;
+  oit: typeof oit;
+  before: typeof before;
+  beforeEach: typeof beforeEach;
+  after: typeof after;
+  afterEach: typeof afterEach;
+  include: typeof include;
+}
+
+type OutcomesType = {
+  PASSED: 'passed';
+  FAILED: 'failed';
+  SKIPPED: 'skipped';
+}
+
+export const Events: EventsType;
+
+export class Harness extends EventEmitter.EventEmitter {
+  constructor(testable: Testable, initial?: zUnitOptions);
+
+  numberOfTests: number;
+
+  report: unknown;
```

Comment:
The report is just a representation of the test suite. More specifically this is a GraphNode object with the following shape...

```
{
 name: "string",
 description: "string",
 point: number,
 parent: A reference to a parent GraphNode (can be undefined)
 children: An array of child GraphNodes
 type: string - One of "suite" or "test"
  result: string - One of "PASSED", "FAILED", "SKIPPED",
  errors: Array of Error objects,
  reason: string,
  exclusive: boolean,
  incomplete: boolean,
  stats: {
    tests: number,
    passed: number,
    failed: number,
    skipped: number,
    duration: number (milliseconds),
  },
}
```

Reply:
Thanks a lot, I've included the `GraphNode` class in the types file.